File: October 2025 GFG SOLUTION/October-25.py
class Solution:
    def minOperations(self, arr):
        import heapq
        s = sum(arr)
        t = s / 2
        h = [-x for x in arr]
        heapq.heapify(h)
        ops = 0
        while s > t:
            x = -heapq.heappop(h)
            s -= x / 2
            heapq.heappush(h, -x / 2)
            ops += 1
        return ops

File: October 2025 GFG SOLUTION/October-25.java
class Solution {
    public int minOperations(int[] arr) {
        double s = 0, t;
        PriorityQueue<Double> q = new PriorityQueue<>(Collections.reverseOrder());
        for (int x : arr) {
            s += x;
            q.offer((double) x);
        }
        t = s / 2;
        int ops = 0;
        while (s > t) {
            double x = q.poll();
            s -= x / 2;
            q.offer(x / 2);
            ops++;
        }
        return ops;
    }
}
